metrics/logpoller_test.go

```diff
@@ -0,0 +1,62 @@
+package metrics
+
+import (
+	"context"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func setupTestLogPollerMetrics(t *testing.T) GenericLogPollerMetrics {
+	m, err := NewGenericLogPollerMetrics("1", "test-network")
+	require.NoError(t, err)
+	return m
+}
+
+func TestLogPollerMetrics_RecordQueryDuration(t *testing.T) {
+	m := setupTestLogPollerMetrics(t)
+	ctx := context.Background()
```

Comment:
nit: for tests think we can just use `t.Context()`

Reply:
Fixed